src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -2262,13 +2262,7 @@ private bool NeedsTypeLayoutCheck(TypeDesc type)
 
         private bool HasLayoutMetadata(TypeDesc type)
         {
-            if (type.IsValueType && (MarshalUtils.IsBlittableType(type) || ReadyToRunMetadataFieldLayoutAlgorithm.IsManagedSequentialType(type)))
-            {
-                // Sequential layout
-                return true;
-            }
-
-            return false;
+            return type is MetadataType metadataType && (metadataType.IsSequentialLayout || metadataType.IsExplicitLayout);
```

Comment:
Can we also delete/replace `IsBlittableType` from `ComputeInstanceFieldLayout` so that the layout does not depend on the algorithm to compute blittability?